dsl_parser/tests/test_get_attributes_list.py

```diff
@@ -0,0 +1,262 @@
+                            inputs:
+                                a: { get_attributes_list: [SOURCE, a] }
+                target_interfaces:
+                    test:
+                        op_with_no_get_attribute:
+                            implementation: p.p
+                            inputs:
+                                a: 1
+                        op_with_get_attribute:
+                            implementation: p.p
+                            inputs:
+                                a: { get_attributes_list: [TARGET, a] }
+"""
+        parsed = prepare_deployment_plan(self.parse(yaml))
+        webserver_node = None
+        for node in parsed.node_templates:
+            if node['id'] == 'webserver':
+                webserver_node = node
+                break
+        self.assertIsNotNone(webserver_node)
+
+        def assertion(operations):
+            op = operations['test.op_with_no_get_attribute']
+            self.assertIs(False, op.get('has_intrinsic_functions'))
+            op = operations['test.op_with_get_attribute']
+            self.assertIs(True, op.get('has_intrinsic_functions'))
+
+        assertion(webserver_node['operations'])
+        assertion(webserver_node['relationships'][0]['source_operations'])
+        assertion(webserver_node['relationships'][0]['target_operations'])
+
+    def test_get_attributes_list(self):
+        node_instances = [
+            {
+                'id': 'goodnode_r1ck00',
+                'node_id': 'goodnode',
+                'runtime_properties': {'sing': 'never'},
+            },
+            {
+                'id': 'badnode_0hn011',
+                'node_id': 'badnode',
+                'runtime_properties': {'sing': 'forget'},
+            },
+            {
+                'id': 'goodnode_45713y',
+                'node_id': 'goodnode',
+                'runtime_properties': {'sing': 'gonna'},
+            },
+            {
+                'id': 'goodnode_r0113d',
+                'node_id': 'goodnode',
+            },
+            {
+                'id': 'goodnode_s1n613',
+                'node_id': 'goodnode',
+                'runtime_properties': {'sing': 'you'},
+            },
+            {
+                'id': 'goodnode_41b2m5',
+                'node_id': 'goodnode',
+                'runtime_properties': {'sing': 'up'},
+            },
+        ]
+        nodes = [
+            {'id': 'badnode'},
+            {'id': 'goodnode', 'properties': {'sing': 'give'}},
+        ]
+        storage = self._mock_evaluation_storage(
+            node_instances=node_instances, nodes=nodes)
+
+        payload = {'a': {'get_attributes_list': ['goodnode', 'sing']}}
+        context = {}
+        result = functions.evaluate_functions(payload.copy(), context, storage)
+
+        attributes_list = result['a']
+        assert len(attributes_list) == 5
+        for value in ('never', 'gonna', 'give', 'you', 'up'):
+            assert value in attributes_list
```

Comment:
if attributes_list is `['up', 'up', 'up', 'up', 'up']` then this test will pass, while it shouldn't have.
I suggest either set()==set(), or just a straight list comparison, which would also assert on the ordering, and I suppose we might as well guarantee ordering within this function?

Reply:
Actually, I'm wrong, the loop is the other way around, you're right! Still, might want to assert on the ordering anyway?

Reply:
True, I'll change it to an all check instead.

I don't think we can guarantee the ordering can we? Under the hood we're retrieving the node_instances list, and I don't think the order is deterministic.

Reply:
Switched to an all() check.

Reply:
But within this function, we can, ie. the guarantee is that "this function returns whatever ordering the backend gives us". I do expect it'll become a question with these functions, as this is pretty much the first list-ordering function we have. So just having the order depend on one thing (essentially, storage-manager) (who does sort by default, on... something. Storage id?) is not bad

Reply:
That's a fair point- OK, I'll change it to that and add a comment to explain that.